### DISCUSSION TITLE
Paraview does not correctly render AMReX plotfiles with large physical geometry

### DISCUSSION THREAD

**Discussion by BenWibking** [AMReX-Codes/amrex#2637] [Q&A]
Hi,

Opening a set of plotfiles produced with my code, I only get a black surface rendering for any of the fields, e.g.:
<img width="1353" alt="Screen Shot 2022-02-28 at 11 47 01 AM" src="https://user-images.githubusercontent.com/1781919/155907666-10e62df6-6ba2-4cef-80f1-0a92e4233fb0.png">

Is this a known issue, or should I open a bug with Paraview for this?

**Comment by asalmgren** [comment]
Could you try reading them with amrvis, yt or visit to make sure the
plotfiles themselves aren’t somehow corrupted?  We use paraview regularly
with no problem.

Sent from my iPhone

On Feb 27, 2022, at 4:49 PM, Ben Wibking ***@***.***> wrote:

﻿

Hi,

Opening a set of plotfiles produced with my code, I only get a black
surface rendering for any of the fields, e.g.:
[image: Screen Shot 2022-02-28 at 11 47 01 AM]
<https://user-images.githubusercontent.com/1781919/155907666-10e62df6-6ba2-4cef-80f1-0a92e4233fb0.png>

Is this a known issue, or should I open a bug with Paraview for this?

—
Reply to this email directly, view it on GitHub
<https://github.com/AMReX-Codes/amrex/discussions/2637>, or unsubscribe
<https://github.com/notifications/unsubscribe-auth/ACRE6YWJNH6TT53FLQLAMBLU5LA77ANCNFSM5PPWFYLA>
.
Triage notifications on the go with GitHub Mobile for iOS
<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
or Android
<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

You are receiving this because you are subscribed to this thread.Message
ID: ***@***.***>

**Reply by BenWibking** [reply]
I can read them in with Visit with no problem. It seems to only happen when the grid sizes exceed 512^3 cells per rank. I've uploaded such a plotfile that exhibits this problem here: https://cloudstor.aarnet.edu.au/plus/s/pM333ijq9pr1QMQ

**Comment by BenWibking** [comment]
The zero-density block is a Paraview bug. It renders correctly in Visit.

-Ben
________________________________

**Comment by etpalmer63** [comment]
I can confirm I was getting strange output for both Paraview 5.9.1 and 5.10. Also values in the spreadsheet view seemed off. 

This is the output I got for Amrvis, does it look correct to you?
![Screenshot from 2022-03-01 11-41-51](https://user-images.githubusercontent.com/24531086/156261523-4ad5bce3-f6eb-44ed-91e7-35f4d9611b5f.png)

**Reply by BenWibking** [reply]
Yeah, that looks right to me.

**Reply by BenWibking** [reply]
There should also be density perturbations on the background, but maybe they are clipped.

**Reply by BenWibking** [reply]
What version(s) of Paraview render correctly in your tests?

**Reply by etpalmer63** [reply]
None :( . I tried 5.9.1 and 5.10.  

The closest I got was this:
![Screenshot from 2022-03-01 10-57-26](https://user-images.githubusercontent.com/24531086/156279963-5f2565e1-f176-498a-8559-037d91ed6dc2.png)

But this clearly has several errors.

**Reply by etpalmer63** [reply]
One issue I noticed in your example was that you have the representation selected as `surface`. Typically I use `volume` for the cell valued plots in 3D. However, when I selected that, it still didn't render properly and caused my machine to crash.

**Reply by BenWibking** [reply]
I usually use surface renderings in VisIt with a slice operator (so it renders faster/nearly interactively), so I was trying to reproduce that here with Paraview.  The motivation for using Paraview is to do real-time/interactive volumetric rendering with NVIDIA IndeX in Paraview (but obviously this bug prevents that from working).

**Reply by BenWibking** [reply]
I've opened a bug on the Paraview Gitlab for these issues: https://gitlab.kitware.com/paraview/paraview/-/issues/21265

**Reply by BenWibking** [reply]
In Paraview 5.10.1, I now get a segmentation fault when trying to read in this file.

**Reply by etpalmer63** [reply]
I tried this again with Paraview 5.10.1 and your data set. I did not get a segmentation fault, instead I got the black surface you previously showed above. It also failed to render as a volume.

**Reply by BenWibking** [reply]
That's odd. I was using Paraview in client/server mode. The segmentation fault happened on the server side.

**Reply by BenWibking** [reply]
Do you know if there anyone from the Paraview team who is able to take a look at fixing this? I haven't been able to get a response to my bug report.

**Reply by BenWibking** [reply]
This is the error message I get on the server side: https://discourse.paraview.org/t/segmentation-fault-on-server-when-reading-amrex-file/9258

**Reply by BenWibking** [reply]
@etpalmer63 @asalmgren The issue is that Paraview can't handle large physical box geometry. If I rescale the geometry so that the RealBox coordinates go from [0, 1] in each dimension, it works fine. I've opened a bug on the Paraview bugtracker for this: https://gitlab.kitware.com/paraview/paraview/-/issues/21371

**Reply by etpalmer63** [reply]
Hi @BenWibking, great info! I'm planning to look into the Paraview code myself and see if I can submit a few fixes. In the meantime, thanks for all your detailed reports on the issues!

**Reply by BenWibking** [reply]
It turns out this is also an issue in Visit's volume rendering (though not slice or surface rendering): https://github.com/visit-dav/visit/issues/17602

**Reply by etpalmer63** [reply]
Hey @BenWibking, since the issues appear to be similar I am wondering whether this comes from the VTK backend. I know ParaView has this, and it does look like VisIt has some of its code. In either case, I went ahead and let both ParaView and VisIt know that the issue occurs in both places since their engineers probably know better what aspects these codes share in common and maybe that will help them troubleshoot. 

Erik

**Reply by BenWibking** [reply]
That makes sense, thanks.
